core/src/main/java/hudson/PluginManager.java

```diff
@@ -1923,6 +1924,31 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
         return FormValidation.ok();
     }
 
+    @Restricted(NoExternalUse.class)
+    @RequirePOST public FormValidation doCheckUpdateSiteUrl(StaplerRequest request, @QueryParameter String value) {
+        if (StringUtils.isNotBlank(value)) {
+            try {
+                value += "?version=" + Jenkins.VERSION;
+
+                URL url = new URL(value);
+
+                // Connect to the URL
+                HttpURLConnection conn = (HttpURLConnection) url.openConnection();
```

Comment:
This method does not properly handle proxy servers. Per its Javadoc, `hudson.ProxyConfiguration#open` should be used wherever `URL#openConnection()` to internet URLs is invoked directly. Please update the code accordingly and describe your testing against a proxy server. To test, I would recommend finding a Docker image that provides a proxy server and starting up a container from that image.

Reply:
Thanks for the suggestion! I've tested it with docker container [datadog/squid](https://hub.docker.com/r/datadog/squid). After setting the proxy in the `HTTP Proxy Configuration` section, the `hudson.ProxyConfiguration#open` works fine to check the connectivity. Have a look at the new commit.